### PR TITLE
[Cluster Launcher] Re Org the cluster launcher pages.

### PR DESCRIPTION
**Pull request by bllchmbrs**
## Why are these changes needed?

We had a fair amount of nesting in the documentation - this surfaces key things like "clouds".

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by anabranch** [comment]
Screen Shot 2020-07-24 at 10 05 26 AM
Screen Shot 2020-07-24 at 10 05 23 AM
Screen Shot 2020-07-24 at 10 05 20 AM
Screen Shot 2020-07-24 at 10 05 13 AM
Screen Shot 2020-07-24 at 10 05 08 AM
Screen Shot 2020-07-24 at 10 05 03 AM

**Review by richardliaw — APPROVED** [review]

File: doc/source/index.rst [hunk 1]
```diff
@@ -142,7 +142,9 @@ Academic Papers
 
    cluster/index.rst
    cluster/launcher.rst
+   cluster/cloud.rst
    cluster/deploy.rst
+   cluster/config.rst
```
Comment (richardliaw) [inline doc/source/index.rst:147]: RE: config to be top level, I'm not sure this is worth it. It's only applicable to the cluster launcher (not for cluster managers or on-prem clusters), and can be easily accessed in the launcher page. The vertical tradeoff doesn't seem worth it.

**Review by richardliaw — APPROVED** [review]

File: doc/source/index.rst (hunk 1 above)
Comment (anabranch) [reply doc/source/index.rst:147]: ok, I will change this. Thanks.

**Comment by anabranch** [comment]
Screen Shot 2020-07-27 at 1 35 26 PM
Screen Shot 2020-07-27 at 1 35 22 PM
Screen Shot 2020-07-27 at 1 35 16 PM
Screen Shot 2020-07-27 at 1 35 11 PM


@richardliaw changed!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**